morphman/automated_landmarking/automated_landmarking_tools.py

```diff
@@ -0,0 +1,245 @@
+##   Copyright (c) Aslak W. Bergersen, Henrik A. Kjeldsberg. All rights reserved.
+##   See LICENSE file for details.
+
+##      This software is distributed WITHOUT ANY WARRANTY; without even
+##      the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+##      PURPOSE.  See the above copyright notices for more information.
+
+from scipy.ndimage.filters import gaussian_filter
+from scipy.signal import argrelextrema
+
+# Local import
+from morphman.common import *
+
+
+def get_maximum_coronal_coordinate(coronal_coordinate, length):
+    value_index = coronal_coordinate[argrelextrema(coronal_coordinate, np.less_equal)[0]].min()
+    max_coronal_coordinate_id = np.array(coronal_coordinate.tolist().index(value_index))
+    search_tolerance = 30
+
+    if abs(length[max_coronal_coordinate_id] - length[-1]) > search_tolerance:
+        print("-- Sanity check failed, checking for maximums")
+
+        value_index = coronal_coordinate[argrelextrema(coronal_coordinate, np.greater_equal)[0]].max()
+        max_coronal_coordinate_id = np.array(coronal_coordinate.tolist().index(value_index))
+        if abs(length[max_coronal_coordinate_id] - length[-1]) > search_tolerance:
+            print("-- Sanity check failed, no anterior bend in model. Exiting.")
+            sys.exit(1)
+    return max_coronal_coordinate_id
+
+
+def get_centerline_coordinates(line, length):
+    x = np.zeros(length.shape[0])
+    y = np.zeros(length.shape[0])
+    z = np.zeros(length.shape[0])
+    for i in range(z.shape[0]):
+        x[i] = line.GetPoints().GetPoint(i)[0]
+        y[i] = line.GetPoints().GetPoint(i)[1]
+        z[i] = line.GetPoints().GetPoint(i)[2]
+
+    return {"x": x, "y": y, "z": z}
+
+
+def orient_centerline(ica_centerline):
+    """
+    Check the orientation of the axial coordinate and
+    reverse centerline if path of centerline is in distal direction.
+
+    Args:
+        ica_centerline (vtkPolyData): Centerline to check
+
+    Returns:
+        ica_centerline (vtkPolyData): Centerline, possibly reversed
+    """
+    length = get_curvilinear_coordinate(ica_centerline)
+    coordinates = get_centerline_coordinates(ica_centerline, length)
+
+    # Iterate through coordinates and find axial coordinate
+    for _, coordinate in coordinates.items():
+        coordinate = gaussian_filter(coordinate, 25)
+        min_c = min(coordinate)
+        max_c = max(coordinate)
+        id0 = coordinate.tolist().index(min_c)
+        id1 = coordinate.tolist().index(max_c)
```

Comment:
Instead of converting to list you could use `np.where`

```python
id0 = np.where(coordinate == np.min(coordinate))[0][0]
id1 = np.where(coordinate == np.max(coordinate))[0][0]
if id0 == 0 and id1 == (coordinate.shape[0] - 1):
```